iree/compiler/Codegen/LLVMCPU/VectorContractCustomKernels.cpp

```diff
@@ -139,8 +143,342 @@ static Value flatten(PatternRewriter &rewriter, Location loc, Value vector) {
   return rewriter.create<vector::ShapeCastOp>(loc, dstType, vector);
 }
 
-/// Converts matrix-times-matrix-transposed vector.contracts with
-/// lhs and rhs inputs defined by arith.extsi promoting from i8 to i32,
+// We only constant-evaluate this function. We could enforce this in C++20 with
+// consteval.
+constexpr bool isPowerOfTwo(int i) { return (i > 0) && ((i & (i - 1)) == 0); }
```

Comment:
Most of such functions exist in `llvm/ADT/MathExtras.h`

Reply:
Thanks! Found `llvm/Support/MathExtras.h`, using it.